miden-tx/src/testing/mock_chain/mod.rs

```diff
@@ -473,53 +458,45 @@ impl MockChain {
         }
 
         self.available_accounts
-            .insert(account.id(), MockAccount::new(account.clone(), seed, authenticator));
+            .insert(account.id(), MockAccount::new(account.clone(), None, authenticator));
 
         MockFungibleFaucet(account)
     }
 
-    /// Adds a new `Account` from an `AccountBuilder` to the list of pending objects.
-    /// A block has to be created to finalize the new entity.
-    pub fn add_from_account_builder(
+    /// Adds the [`AccountComponent`] corresponding to `auth_method` to the account in the builder
+    /// and builds a new or existing account depending on `account_state`.
+    ///
+    /// This account is added to the available accounts and are immediately available without having
+    /// to seal a block.
+    fn add_from_account_builder(
         &mut self,
         auth_method: Auth,
-        account_builder: AccountBuilder,
+        mut account_builder: AccountBuilder,
+        account_state: AccountState,
     ) -> Account {
-        let (account, seed, authenticator) = match auth_method {
-            Auth::BasicAuth => {
-                let mut rng = ChaCha20Rng::from_seed(Default::default());
-                let sec_key = SecretKey::with_rng(&mut rng);
-                let pub_key = sec_key.public_key();
-
-                let (acc, seed) = account_builder
-                    .with_component(RpoFalcon512::new(pub_key))
-                    .build_testing()
-                    .unwrap();
-
-                let authenticator = BasicAuthenticator::<ChaCha20Rng>::new_with_rng(
-                    &[(pub_key.into(), AuthSecretKey::RpoFalcon512(sec_key))],
-                    rng,
-                );
-
-                (acc, seed, Some(authenticator))
-            },
-            Auth::NoAuth => {
-                let (account, seed) = account_builder.build_testing().unwrap();
-                (account, seed, None)
+        let authenticator = match Self::build_authentication_component(auth_method) {
+            Some((auth_component, authenticator)) => {
+                account_builder = account_builder.with_component(auth_component);
+                Some(authenticator)
             },
+            None => None,
+        };
+
+        let (account, seed) = if let AccountState::New = account_state {
+            account_builder.build().map(|(account, seed)| (account, Some(seed))).unwrap()
+        } else {
+            account_builder.build_existing().map(|account| (account, None)).unwrap()
         };
 
         self.available_accounts
             .insert(account.id(), MockAccount::new(account.clone(), seed, authenticator));
 
-        self.add_account(account.clone());
```

Comment:
Why do we no longer need this?

Reply:
Perhaps @igamigo can clarify this. It's not clear to me when adding new accounts in the chain whether they must be added to both available_accounts and pending_objects or only the former.

It seems the former is sufficient, so since we're adding the account to available_accounts adding it to pending_objects seems unnecessary. And we're already not doing it for the "existing faucet" case, so I made it consistent: https://github.com/0xPolygonMiden/miden-base/blob/6e0e29f477e08ffbd5900ccb951818d1cecab13b/miden-tx/src/testing/mock_chain/mod.rs#L475-L479

Reply:
Originally the intent was to keep `MockChain` consistent with how the actual chain would evolve, (currently not entirely the case since some updates are not performed entirely as they should) considering things like updates to the account and note trees. For new accounts, I don't think it's needed to add the account to pending objects, but I believe I originally added it for convenience (since it did not hurt either).